tensorboard/plugins/custom_scalar/tf_custom_scalar_dashboard/tf-custom-scalar-margin-chart-card.html

```diff
@@ -518,9 +560,26 @@ <h1>[[_titleDisplayString]]</h1>
             }
           });
 
+          const priorListOfTags = this._tagsToMissingRuns[run];
           if (tagsNotFound.length) {
-            // At least 1 tag could not be found. Show an error message.
-            this.set('_tagsWithNoData', tagsNotFound);
+            // At least 1 tag could not be found. Show a warning message.
+            if (!_.isEqual(priorListOfTags, tagsNotFound)) {
+              this._tagsToMissingRuns[run] = tagsNotFound;
+              // We trigger notifications in polymer by creating a new object.
+              // Polymer's property observers do not work for properties with
+              // names containing periods, and run names may have periods.
```

Comment:
Hmm, it's a bit unfortunate to not be able to use the native mutation notification logic.  I wonder if we could work around this by instead storing `enumeratedMissingTags` directly (so in contravention of my comment above) and then doing the mutations on that using the built-in array mutation methods.

The only caveat would be not being able to look up a run's missing tag entry directly by the run name (one would instead have to iterate through the array).  But we could fix that by just keeping a separate `runToMissingTagsIndex` property that maps run names to indexes in the array.  (In fact, I think we could even technically use the same property object by just adding the runs as properties on the array object, though maybe that's a little too hacky.)

Reply:
Using array mutations works!